### PR TITLE
Nodes History endpoint

### PR DESCRIPTION
**Pull request by Mykhailenko**
Group by node source name, then hostname, then node name.
Sort by start time.
If start time before window then set it to window start.
if end time after window (or zero) then set it to window end.

### CODE REVIEW

File: rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRestInterface.java [hunk 1]
```diff
@@ -765,4 +766,10 @@ String getStatHistory(@HeaderParam("sessionid") String sessionId, @QueryParam("r
     String getNodeThreadDump(@HeaderParam("sessionid") String sessionId, @QueryParam("nodeurl") String nodeUrl)
             throws NotConnectedException;
 
+    @GET
+    @GZIP
+    @Path("nodes/history")
+    @Produces("application/json")
+    List<RMNodeHistory> getNodesHistory(@HeaderParam("sessionid") String sessionId, @HeaderParam("windowStart") long windowStart,
```
Comment (aminelouati) [inline rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRestInterface.java]: Very good job !
I suggest to return a Map<String, List<RMNodeHistory>> to ease the work from the client side.

Comment (Mykhailenko) [reply rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRestInterface.java]: good. i will do it

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/db/RMDBManager.java [hunk 2]
```diff
@@ -814,4 +814,14 @@ public RMDBManagerBuffer getBuffer() {
         return rmdbManagerBuffer;
     }
 
+    public List<NodeHistory> getNodesHistory(long windowStart, long windowEnd) {
+        return executeReadTransaction(session -> {
+            Query query = session.createQuery("FROM NodeHistory " +
```
Comment (fviale) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/db/RMDBManager.java]: endtime is by default null, how will this query behave for :windowStart <= endTime and endTime <= :windowEnd

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/db/RMDBManager.java [hunk 3]
```diff
@@ -814,4 +814,21 @@ public RMDBManagerBuffer getBuffer() {
...
     }
 
+    public List<NodeHistory> getNodesHistory(long windowStart, long windowEnd) {
+        return executeReadTransaction(session -> {
+            //            this expression is built as negation to expression that
+            //            retrieves all items that we do not want to have in the response.
+            //            just to clarify the events we include
+            //            we discard only events :
+            //            - ending before the windows start
+            //            - starting after the window start
```
Comment (aminelouati) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/db/RMDBManager.java]: starting after the window End ?

Comment (Mykhailenko) [reply rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/db/RMDBManager.java]: yes you are right

**Pull request merged**